apps/admin_panel/assets/src/omg-account-current/action.js

```diff
@@ -1,38 +1,55 @@
-
 import { setCurrentAccount } from '../services/sessionService'
 import * as settingService from '../services/settingService'
 import * as accountService from '../services/accountService'
-export const loadCurrentAccount = accountId => async dispatch => {
+export const loadCurrentAccount = accountId => async (dispatch, getState, { socket }) => {
   dispatch({ type: 'CURRENT_ACCOUNT/REQUEST/INITIATED' })
   try {
     const result = await accountService.getAccountById(accountId)
     if (result.data.success) {
-      return dispatch({ type: 'CURRENT_ACCOUNT/REQUEST/SUCCESS', currentAccount: result.data.data })
+      socket.joinChannel(`account:${result.data.data.id}`)
+      dispatch({ type: 'CURRENT_ACCOUNT/REQUEST/SUCCESS', currentAccount: result.data.data })
     } else {
-      return dispatch({ type: 'CURRENT_ACCOUNT/REQUEST/FAILED', error: result.data.data })
+      dispatch({ type: 'CURRENT_ACCOUNT/REQUEST/FAILED', error: result.data.data })
     }
+    return result
   } catch (error) {
+    console.log('Failed to load current account with error', error)
     return dispatch({ type: 'CURRENT_ACCOUNT/REQUEST/FAILED', error })
   }
 }
 
-export const updateCurrentAccount = ({ accountId, name, description, avatar }) => async dispatch => {
+export const updateCurrentAccount = ({
+  accountId,
+  name,
+  description,
+  avatar
+}) => async dispatch => {
   try {
-    const resultUpdateAccount = await settingService.updateAccountInfo({ id: accountId, name, description })
+    const resultUpdateAccount = await settingService.updateAccountInfo({
```

Comment:
Can you simplified conditional logic? also make `resultUpdateAccount.data.data` more understandable (maybe assigned it to new variable with descriptive name `const updatedAccount = resultUpdateAccount.data.data`).